### ISSUE TITLE
options for method with Options => true

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#1881]
@MichaelABurr suggested that `options` for a method with `Options => true` should list all installed methods and their options, rather than null.

**Comment by DanGrayson** [comment]
Alternatively, one could make `options n` work, when n is an integer:
```m2

i2 : methods dual

o2 = {0 => (dual, BettiTally)                }
     {1 => (dual, ChainComplex)              }
     {2 => (dual, ChainComplexMap)           }
     {3 => (dual, CoherentSheaf)             }
     {4 => (dual, Matrix)                    }
     {5 => (dual, Module)                    }
     {6 => (dual, MonomialIdeal)             }
     {7 => (dual, MonomialIdeal, List)       }
     {8 => (dual, MonomialIdeal, RingElement)}

o2 : NumberedVerticalList

i3 : options o2#7

o3 = OptionTable{Strategy => 0}

o3 : OptionTable

i4 : options 7
stdio:4:1:(3): error: no method found for applying options to:
     argument   :  7 (of class ZZ)
```

**Comment by DanGrayson** [comment]
One could also make this work:
```m2

i5 : options methods dual
stdio:5:1:(3): error: no method found for applying options to:
     argument   :  {0 => (dual, BettiTally)   . (of class NumberedVerticalList)
                   {1 => (dual, ChainComplex) .
                   {2 => (dual, ChainComplexMa.
                   {3 => (dual, CoherentSheaf).
                   ............................
```

**Comment by MichaelABurr** [comment]
I like the idea of making   options n    work with some meaningful output for    options dual     to suggest where to look (instead of no output).

**Comment by mahrud** [comment]
> I like the idea of making   options n    work with some meaningful output for    options dual     to suggest where to look (instead of no output).

I agree. In particular, if the user doesn't already know that `dual` has `Options => true`, they wouldn't know that `options dual` isn't complete.

**Comment by DanGrayson** [comment]
Good point.

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

Carrying over a second ask from #2864, which is being closed as a duplicate of this issue: the
`options`-returns-null half of it is this issue, but @mikestillman raised something there that is
not recorded here.

> Similarly, if we write a version of dual without optional arguments, currently we need to write
> something like `dual MyType := {} >> o -> F -> (....)`. However, since the code can know that
> `dual` takes options, if none are given, perhaps the user shouldn't need to write the
> `{} >> o ->` part?

Still true on 1.26.06, and the failure is silent. Omitting the boilerplate is accepted, and the
function then receives the pair `(OptionTable{}, MyType{})` rather than the object:

```m2
i3 : dual MyType := F -> class F

i4 : dual X

o4 = Sequence
```

It also swallows unknown options, where the `{} >> o ->` form rejects them.

The useful part for whoever takes it on: **the shortcut already exists for a different dispatch
mode, ten lines away.** `SingleWithOptions` (`m2/methods.m2:141`) and the binary path (`:103`) both
short-circuit with `if #o === 0 then (f -> f arg)`, so a plain function works there —

```m2
i5 : f = method(Options => true, Dispatch => Thing);

i6 : f MyType := F -> class F

i7 : f X

o7 = MyType
```

— while `MultipleArgsWithOptions` (`:152-163`), which is what `dual` and every default-dispatch
method with options is, always calls `innerMethodFunction(o, arg)`.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>
